### PR TITLE
[Perf] Skip creating attention mask in llama dataloader

### PR DESCRIPTION
**Pull request by billishyahao**
This patch is to skip creating attention mask in llama dataloader by adding flag **--no-create-attention-mask-in-dataloader**. With this patch, we can see multiple benefits below:
1) This could bring 4%~6% performance gain
2) Also address observed data loader crash issue when dealing with long sequence case e.g. https://github.com/NVIDIA/Megatron-LM/issues/1025
3) We also see new megatron model example adopt this flag as well. e.g. https://github.com/NVIDIA/Megatron-LM/blob/40db706d37a25787b0fb6b7b561327e5d2b4b2e4/examples/mamba/train.sh#L102

### CODE REVIEW

**Comment by wenchenvincent** [comment]
@lizamd Are you aware of this change? Do we use this setting for testing?

**Comment by wenchenvincent** [comment]
@billishyahao Could you give some more details on the behavior `--no-create-attention-mask-in-dataloader`? For example, if attention mask is not created in dataloader, where is it created?

**Comment by lizamd** [comment]
@billishyahao could you provide more data on the 4-5% perf gain and address @wenchenvincent 's question? we can have a call too

**Comment by zstreet87** [comment]
this sounds like something an IFU would pick up, right? @billishyahao is this still needed?

**Comment by zstreet87** [comment]
@billishyahao please address my last comment

**Comment by zstreet87** [comment]
closing as author isn't responding. You can always reopen and address my previous comments.
